subsys/fs/ext2/ext2_impl.c

```diff
@@ -688,10 +688,6 @@ ssize_t ext2_inode_write(struct ext2_inode *inode, const void *buf, uint32_t off
 
 int ext2_inode_trunc(struct ext2_inode *inode, off_t length)
 {
-	if (length > UINT32_MAX) {
```

Comment:
Question: If you compile the code for native_posix AMD64, wouldn't that make `off_t` `long long`, aka `int64_t`?

Reply:
```
int main(void)
{
        printf("Hello World! %s\n", CONFIG_BOARD);
        printf("%u\n", sizeof(off_t));
        return 0;
}
```

```$ ./build/zephyr/zephyr.elf 
*** Booting Zephyr OS build zephyr-v3.5.0-2011-g716de6fef2c6 ***
Hello World! native_posix
4
```

```
$ lscpu | head -n1
Architecture:                    x86_64
```

Reply:
Which is curious because it is `int64_t` outside of Zephyr.

Reply:
Same change

```
west build -b native_posix_64 -d delete-me samples/hello_world/
...
$$ delete-me/zephyr/zephyr.elf
*** Booting Zephyr OS build zephyr-v3.4.0-7401-g716de6fef2c6 ***
Hello World! native_posix_64
8
```

Reply:
> Which is curious because it is `int64_t` outside of Zephyr.

Probably Zephyr env does not set _FILE_OFFSET_BITS for for sdk?

Reply:
ack, I wasn't aware there is a separate native_posix_64 board.

I assume the next step would be to mark this bug as false-positive?

Reply:
Confusing as meaning of WORD in assembly.

Reply:
> ack, I wasn't aware there is a separate native_posix_64 board.
> 
> I assume the next step would be to mark this bug as false-positive?

Yes.

Reply:
Thanks